cloudsmith_cli/cli/tests/commands/policy/test_licence.py

```diff
@@ -0,0 +1,181 @@
+
+    if not headers:
+        raise Exception("Table not found in output!")
+
+    if not row:
+        raise Exception("No license policies found - expected 1.")
+
+    return dict(zip(headers, row))
+
+
+def assert_output_matches_policy_config(output, config_file_path):
+    """Assert that tabular output from a command invocation matches policy config."""
+
+    output_table = parse_table_from_output(output)
+    config = json.loads(config_file_path.read_text())
+
+    # Assert that configurable values are set correctly
+    assert output_table["Name"] == config["name"]
+    assert output_table["Description"] == config["description"]
+    assert output_table["SPDX Identifiers"] == str(config["spdx_identifiers"])
+    assert (
+        output_table["Allow Unknown Licenses"]
+        == str(config["allow_unknown_licenses"]).lower()
+    )
+    assert (
+        output_table["Quarantine On Violation"]
+        == str(config["on_violation_quarantine"]).lower()
+    )
+
+    # We just require non-configurable values to be truthy
+    assert output_table["Created"]
+    assert output_table["Updated"]
+    assert output_table["Identifier"]
+
+    # Return the slug_perm in case we need it for other cli calls
+    return output_table["Identifier"]
+
+
+@pytest.mark.usefixtures("set_api_key_env_var", "set_api_host_env_var")
+def test_license_policy_commands(runner, organization, tmp_path):
+    """Test CRUD operations for license policies."""
+
+    # Generate the license policy configuration file.
+    policy_name = random_str()
+
+    policy_config_file_path = create_license_policy_config_file(
+        directory=tmp_path,
+        name=policy_name,
+        description=random_str(),
+        allow_unknown_licenses=random_bool(),
+        on_violation_quarantine=random_bool(),
+        spdx_identifiers=["Apache-2.0"],
+    )
+
+    # Create the license policy
+    result = runner.invoke(
+        create,
+        args=[organization, str(policy_config_file_path)],
+        catch_exceptions=False,
+    )
+    assert (
+        "Creating " + policy_name + " license policy for the cloudsmith namespace ...OK"
+        in result.output
+    )
+    assert "Results: 1 license policy" in result.output
+    slug_perm = assert_output_matches_policy_config(
+        result.output, policy_config_file_path
+    )
+
+    # Use the cli to get the policy
+    result = runner.invoke(ls, args=[organization], catch_exceptions=False)
+    assert "Getting license policies ... OK" in result.output
+    # Ignoring the exact number of results and pluralization of the word "policy"
+    # here, because we may be in a system with pre-existing policies.
+    assert "Results:" in result.output
+    assert_output_matches_policy_config(result.output, policy_config_file_path)
+
+    # Change the values in the config file
+    policy_config_file_path = create_license_policy_config_file(
+        directory=tmp_path,
+        name=policy_name,
```

Comment:
If `policy_name` isn't used in the rest of this test, change this to `name=random_str()`